include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -828,6 +866,17 @@ def StatePreparation : Pass<"state-prep", "mlir::ModuleOp"> {
   ];
 }
 
+def StateValidation : Pass<"state-validation", "mlir::ModuleOp"> {
+  let summary =
+    "Make sure MLIR is valid after synthesis for quantum devices";
+  let description = [{
+    Argument synthesis should replace all `quake.init` from state instructions
+    and calls to state-related runtime functions.
+    Make sure none of them left, and remove definitions for state-related
```

Comment:
If the definitions are marked `private` there is nothing special to do. `private` artifacts that are unreferenced will be cleaned up automatically.

Reply:
Some later ops are failing because those definitions are still in the MLIR... Will double check if they are private.

Reply:
Added SymbolDCE opt after synthesis and cleanup opts, it worked, thanks! Removed the StateValidation pass entirely.